src/app/features/trade/components/mev-bot/mev-bot.component.ts

```diff
@@ -13,9 +13,17 @@ export class MevBotComponent {
 
   public displayMev: boolean = false;
 
-  @Input() set tradePrice(value: BigNumber | undefined) {
+  @Input() set trade(trade: CrossChainTrade | OnChainTrade) {
     const minDollarAmountToDisplay = 999;
-    this.displayMev = value ? value.gt(minDollarAmountToDisplay) : false;
+    const amount = trade?.from.price.multipliedBy(trade?.from.tokenAmount);
+    const isOnChainSwap = trade?.from.blockchain === trade?.to.blockchain;
+
+    if (isOnChainSwap) {
+      this.displayMev = false;
+    } else {
+      this.displayMev = amount ? amount.gt(minDollarAmountToDisplay) : false;
```

Comment:
this.displayMev = amount && isCrossChain ? amount.gt(minDollarAmountToDisplay) : false;